x/did/client/cli/txDID.go

```diff
@@ -0,0 +1,331 @@
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+
+			did, err := types.ParseDID(args[0])
+			if err != nil {
+				return err
+			}
+			verificationMethodID, err := types.ParseVerificationMethodID(args[1], did)
+			if err != nil {
+				return err
+			}
+			privKey, err := getPrivKeyFromKeyStore(verificationMethodID, inBuf)
+			if err != nil {
+				return err
+			}
+
+			// For proving that I know the private key. It signs on the DID, not DIDDocument.
+			signableDID := types.SignableDID(did)
+			sig, err := signUsingCurrentSeq(clientCtx, did, privKey, signableDID)
+			if err != nil {
+				return err
+			}
+
+			fromAddress := clientCtx.GetFromAddress()
+			msg := types.NewMsgDeactivateDID(did, verificationMethodID, sig, fromAddress.String())
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), &msg)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+// readBIP39ParamsFrom reads a mnemonic and a bip39 passphrase from the reader in the interactive mode.
+// It returns empty strings in the non-interactive mode, so that they can be auto-generated by crypto.GenSecp256k1PrivKey.
+func readBIP39ParamsFrom(interactive bool, reader *bufio.Reader) (string, string, error) {
+	if !interactive {
+		return "", "", nil
+	}
+
+	// mnemonic can be an empty string
+	mnemonic, err := input.GetString("Enter your BIP39 mnemonic, or hit enter to generate one:", reader)
+	if err != nil {
+		return "", "", err
+	}
+	if mnemonic != "" && !bip39.IsMnemonicValid(mnemonic) {
+		return "", "", fmt.Errorf("invalid mnemonic")
+	}
+
+	// passphrase can be an empty string
+	passphrase, err := input.GetString("Enter your BIP39 passphrase, or hit enter:", reader)
+	if err != nil {
+		return "", "", err
+	}
+	if passphrase != "" {
+		repeat, err := input.GetString("Repeat the passphrase:", reader)
+		if err != nil {
+			return "", "", err
+		}
+		if passphrase != repeat {
+			return "", "", fmt.Errorf("passphrases don't match")
+		}
+	}
+
+	return mnemonic, passphrase, nil
+
+}
+
+func keystoreBaseDir() string {
+	return filepath.Join(viper.GetString(cli.HomeFlag), "did_keystore")
+}
+
+// savePrivKeyToKeyStore saves a privKey using a password which is read from the reader.
+func savePrivKeyToKeyStore(verificationMethodID string, privKey secp256k1.PrivKey, reader *bufio.Reader) error {
+	passwd, err := getCheckPassword(reader)
+	if err != nil {
+		return err
+	}
+	ks, err := didcrypto.NewKeyStore(keystoreBaseDir())
+	if err != nil {
+		return err
+	}
+	_, err = ks.Save(verificationMethodID, privKey[:], passwd)
+	return err
+}
+
+// Deprecated to https://github.com/cosmos/cosmos-sdk/pull/5904/commits/c16d93e90d6a698cead7c19b55fcede44587aa7f
+func getCheckPassword(reader *bufio.Reader) (string, error) {
```

Comment:
This method deprecated in cosmos sdk.
So it was implemented out of necessity.
https://github.com/cosmos/cosmos-sdk/pull/5904/commits/c16d93e90d6a698cead7c19b55fcede44587aa7f